### DISCUSSION TITLE
Can't configure nvim-java for LunarVim

### DISCUSSION THREAD

**Discussion by skryvets** [nvim-java/nvim-java#288] [Q&A]
I'm trying to set up nvim-java in LunarVim, but I've run into an issue:

```
Failed
◍ jdtls
  ▼ Displaying full log
    Downloading file "https://download.eclipse.org/jdtls/milestones/1.37.0/jdt-language-server-1.37.0-202408011337.tar.gz"…
    spawn: wget failed with exit code 8 and signal 0. 
    Failed to download file "https://download.eclipse.org/jdtls/milestones/1.37.0/jdt-language-server-1.37.0-202408011337.tar.gz".
```

Here's what I added to my ~/.config/lvim/config.lua:

```lua
--
{
  'nvim-java/nvim-java',
  config = function()
    require('java').setup({
      root_markers = {
        'settings.gradle',
        'settings.gradle.kts',
        'pom.xml',
        'build.gradle',
        'build.gradle.kts',
        'mvnw',
        'gradlew',
      },
    })
    require('lspconfig').jdtls.setup({})
  end
},
{ 'nvim-lspconfig/nvim-lspconfig' },
{ 'MunifTanjim/nui.nvim' },
{ 'nvim-java/lua-async' },
{ 'nvim-java/nvim-java-refactor' },
{ 'nvim-java/nvim-java-dap' },
{ 'nvim-java/nvim-java-core' },
{ 'nvim-java/nvim-java-test' },
--
```

JDTLS error disappears when I comment out the `require('java').setup(...)` portion, but then the plugin doesn't function as expected.

I've tried looking for solutions in the following issues:

- https://github.com/williamboman/mason.nvim/issues/1538
- https://github.com/williamboman/mason.nvim/issues/1508

I also attempted modifying the `~/.local/share/nvim/mason/registries/github/mason-org/mason-registry/registry.json` file, but Mason seems to revert the changes with its configuration.

I came across a suggestion that mentioned setting up a custom registry for the packages, but I don't fully understand how to implement this.

The biggest puzzle is that I tried this on AstroVim, and it works fine out of the box. I know LunarVim hasn't been updated lately, but I have quite a few configurations tied to it, and not in a position to switch to another distro (using this professionally). Could you please guide me in the right direction as to why Mason is trying to download a non-existent jdtls version? Also, LunarVim's default jdtls works fine out of the box, but it doesn't include all the goodies that the nvim-java plugin has. 

I would really appreciate any guidance, especially if there's something I'm overlooking in my setup or if there's a better way to handle the jdtls integration in LunarVim.

Thanks in advance!

**Comment by skryvets** [comment]
After I did a bit of research it seems like it's taking 1.37 from one place and timestamp from another. The full version - https://download.eclipse.org/jdtls/milestones/1.37.0/jdt-language-server-1.37.0-202408011337.tar.gz doesn't exists. 

I proposed a PR - https://github.com/nvim-java/mason-registry/pull/21

**Comment by s1n7ax** [comment]
@skryvets This happens due to incorrectly configured mason.nvim
https://github.com/nvim-java/nvim-java/wiki/Q-&-A#no_entry-cannot-find-package-xxxxx

**Comment by skryvets** [comment]
@s1n7ax, thank you so much for looking at it!

I applied (updated from the initial one I had) the following config (which uses opts instead of function), but still getting the same error:

```lua
{
    "nvim-java/nvim-java",
    lazy = false,
    dependencies = {
      "nvim-java/lua-async-await",
      "nvim-java/nvim-java-core",
      "nvim-java/nvim-java-test",
      "nvim-java/nvim-java-dap",
      "nvim-java/nvim-java-refactor",
      "MunifTanjim/nui.nvim",
      "neovim/nvim-lspconfig",
      "mfussenegger/nvim-dap",
      {
        "williamboman/mason.nvim",
        opts = {
          registries = {
            "github:nvim-java/mason-registry",
            "github:mason-org/mason-registry",
          },
        },
      },
    },
    config = function()
      require('java').setup({
        root_markers = {
          'settings.gradle',
          'settings.gradle.kts',
          'pom.xml',
          'build.gradle',
          'build.gradle.kts',
          'mvnw',
          'gradlew',
        },
      })
      require("lspconfig").jdtls.setup {}
    end,
  }
```

Any chance you can provide some tips?

Again, thank you so much for looking at it.

**Reply by candeemis** [reply]
I applied [the configs shared](https://github.com/orgs/nvim-java/discussions/296#discussioncomment-10505201) @fmorroni by on top of yours, and worked for me. 🤔 Thanks for sharing by the way.

**Comment by s1n7ax** [comment]
Is your config in github?
